From 734fa8f7b730db7c7bc75ef4179b2304e73761fc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sun, 16 Feb 2025 22:07:07 +0100
Subject: [PATCH] test: remove the __extension__ keyword  in
 filter_complex_blocks

This keyword is used by GCC and Clang to prevent -Wpedantic (and other
options) from emitting warnings for many GNU C extensions. This is used
heavily in Alpine Linux through musl libc and foritfy-headers. Without
filtering the __extension__ keyword some type definitions are
duplicated. For example, timeval is defined once as

  struct timeval { time_t tv_sec; suseconds_t tv_usec; };

and once as:

  __extension__ struct timeval { time_t tv_sec; suseconds_t tv_usec; };

Without this patch, the LuaJIT C parser doesn't recognize that these
definitions are equivalent, causing unit test to fail on Alpine Linux.
---
 test/unit/testutil.lua | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/unit/testutil.lua b/test/unit/testutil.lua
index a6db7beab1..627f30aadb 100644
--- a/test/unit/testutil.lua
+++ b/test/unit/testutil.lua
@@ -151,6 +151,9 @@ local function filter_complex_blocks(body)
         or string.find(line, 'mach_vm_range_recipe')
       )
     then
+      -- Remove GCC's extension keyword which is just used to disable warnings.
+      line = string.gsub(line, '__extension__', '')
+
       result[#result + 1] = line
     end
   end
